int64 cx
int64 cy
int64 label